src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentMarkupEncodingPass.cs

```diff
@@ -38,17 +41,18 @@ private class Rewriter : IntermediateNodeWalker
             // We want to use AddMarkupContent to avoid aggresive encoding during prerendering.
             // Specifically, when one of the following characters are in the content,
             // 1. New lines (\r, \n), tabs(\t) - so they get rendered as actual new lines, tabs instead of &#xA;
-            // 2. Ampersands (&) - so that HTML entities are rendered correctly without getting encoded
-            // 3. Any character outside the ASCII range
+            // 2. Any character outside the ASCII range
 
-            private static readonly char[] EncodedCharacters = new[] { '\r', '\n', '\t', '&' };
+            private static readonly char[] EncodedCharacters = new[] { '\r', '\n', '\t' };
+
+            private readonly HashSet<string> _seenEntities = new HashSet<string>(StringComparer.Ordinal);
```

Comment:
This is not thread-safe

Reply:
It is in this case because I create a new `Rewriter` for every call to `ExecuteCore()`.